react/shippingStateMachine/useShippingStateMachine.ts

```diff
@@ -72,34 +79,54 @@ const useShippingStateMachine = ({
       }),
     },
     services: {
-      tryToEditReceiverInfo: (ctx, { receiverName }) => {
+      tryToEditReceiverInfo: async (ctx, event) => {
+        if (event.type !== 'SUBMIT_RECEIVER_INFO') {
+          return
+        }
+
         return updateSelectedAddress({
           ...ctx.selectedAddress,
-          receiverName,
+          receiverName: event.receiverName,
         })
       },
-      tryToCreateAddress: (_, event) => {
+      tryToCreateAddress: async (_, event) => {
+        if (event.type !== 'SUBMIT_CREATE_ADDRESS') {
+          return
+        }
```

Comment:
Noob question: why is this assertion needed? Shouldn't this be responsibility of whoever calls this event?

Reply:
not necessarily, you can invoke this service from any event, but it should only be executed on this specific event because it has the `address` parameter which doesn't exist on other events. basically, type checking haha